cityofaustin/atd-moped
# Add safe routes to school plan recommendations to component descriptions

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/14476

This PR adds a script to add SRTS info from a Google sheet-exported csv file to project component descriptions by matching up SRTS IDs associated with both. Multiple project components can have the same SRTS ID.

## Testing
**URL to test:** 

n/a

**Steps to test:**
1. Follow the instructions in the readme added in this PR to get ready to run this script locally
2. When running the script, comment out line 142 (sleep) - otherwise it will take up to an hour to complete ⏳ 
3. You should see the following output (or similar depending on the prod data dump used)
```
Found 4673 rows in csv file.
Fetching components matching collected SRTS IDs.
Found 6486 components to update. Updating...
Done.

Updated 6486 components.
Found 12 SRTS IDs in csv with no match in database: ['1J - 901', '4C - 008', '3G - 901', '3B - 214', '4G - 002', '5A - 003', '2E - 032', '2K - 626', '3F - 901', 'EarlyOut-05', '3C - 004b', '3G - 016b']
Errors on project component IDs: []
```
4. To confirm that the SRTS info was added to records, look at the following `project_component_id`s in `moped_proj_components` in your local database:
   - id# 104 (example of record with `NULL` for existing description replaced with SRTS by this script)
   - id# 118 (example of record with empty string for existing description replaced with SRTS by this script)
   - id# 180 (example of record with existing description and SRTS info appended by this script)
5. As another way of testing, I modified the script to append an easily searchable string like `abc123` to each component description, and I confirmed that the number of rows that contain that string matches the number of rows modified by the script. You can check it out that way too if you'd like.
6. Last, check the activity log of the projects listed earlier in the steps. You should see updates credited to JD. These will be credited to **Data and Technology Admin** when this script is run on production. See [this discussion](https://austininnovation.slack.com/archives/CNUEPKLB1/p1701283671241449) for more about that.
---
#### Ship list
